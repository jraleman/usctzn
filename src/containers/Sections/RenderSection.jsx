import React from 'react';
import styled from 'styled-components';
import FlashCard from '../../components/FlashCard';
import uniqueId from '../../utils/helpers/uniqueId';

const RenderSection = ({ title = '', content = [] }) => (
    <SectionContainer>
        <SectionTitle>{title}</SectionTitle>
        {content.map((c) => (
            <FlashCard
                key={uniqueId('flash-card_')}
                question={Object.keys(c)}
                answers={Object.values(c)[0]}
            />
        ))}
        <SectionDivider />
    </SectionContainer>
);

const SectionTitle = styled.summary`
    text-align: center;
    text-transform: uppercase;
    color: ${({ theme }) => theme.color.text};
    &:hover {
        cursor: pointer;
    }
`;

const SectionContainer = styled.details`
    padding: 2em;
`;

const SectionDivider = styled.hr``;

export default RenderSection;
