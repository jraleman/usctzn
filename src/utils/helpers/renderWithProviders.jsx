import React from 'react';
import { render } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import { lightTheme } from '../styles/theme';

const renderWithProviders = (Component) => {
    render(
        <ThemeProvider theme={lightTheme}>
            {Component}
        </ThemeProvider>
    )
};

export default renderWithProviders;
