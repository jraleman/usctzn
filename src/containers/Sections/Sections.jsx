import React from 'react';
import styled from 'styled-components';
import RenderSection from './RenderSection';
import uniqueId from '../../utils/helpers/uniqueId';

const Sections = ({ data = [] }) => {
    const title = "Questions and Answers";
    return (
        <SectionsContainer>
            <Title>{title}</Title>
            {data.map(({ title, content }) => (
                <RenderSection
                    key={uniqueId('section_')}
                    title={title}
                    content={content}
                />
            ))}
        </SectionsContainer>
    );
};

const Title = styled.h2`
    text-align: center;
    text-transform: uppercase;
`;

const SectionsContainer = styled.div``;

export default Sections;
