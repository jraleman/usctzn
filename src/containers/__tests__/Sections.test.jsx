import React from 'react';
import Sections from '../Sections';
import renderWithProviders from '../../utils/helpers/renderWithProviders';
import americanGovermentData from '../../data/american-goverment.json';
import systemOfGovermentData from '../../data/system-of-goverment.json';

describe('Sections', () => {
    it('renders', async () => {
        const container = renderWithProviders(<Sections />);
        expect(container).toMatchSnapshot();
        expect(container).not.toBeNull();
    });

    it('shows some sections', async () => {
        const data = [
            americanGovermentData,
            systemOfGovermentData
        ];
        const component = renderWithProviders(<Sections data={data} />);
        expect(component).not.toBeNull();
        expect(component).toMatchSnapshot();
    });
});
