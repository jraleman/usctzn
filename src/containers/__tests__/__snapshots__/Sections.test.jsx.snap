// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sections renders 1`] = `undefined`;

exports[`Sections shows some sections 1`] = `undefined`;
